### ISSUE TITLE
NEURAL ARCHITECTURE SEARCH WITH REINFORCEMENT LEARNING

### ISSUE THREAD

**Issue by ghuijo** [One-Day-One-Paper/One-Day-One-Paper#335]
*NAS 논문*

* NAS 구조
  <image src="https://user-images.githubusercontent.com/67627471/206909097-436cd00a-d51e-452e-a406-ede08f14bd41.png" width=50% height=50%>
* the controller RNN finishes generating an architecture --> a neural network with this architecture is built and trained
  <image src="https://user-images.githubusercontent.com/67627471/206910098-cc3640af-f6f4-49f1-9c6b-38136d246377.png" width=60% height=60%>
* the accuracy of the network on a held-out validation set is recorded --> the parameters of the controller RNN are optimized
* describe a policy gradient method which we use to update parameters θc
* the reward signal R is non-differentiable
  * need to use a policy gradient method to iteratively update θc
  * use the REINFORCE rule
* accelerate training with parallelism and asynchronous updates
  <image src="https://user-images.githubusercontent.com/67627471/206909921-13d5ce9a-99f0-4a9f-b34f-77c2535d024b.png" width=70% height=70%>
  * have a parameter server of S shards, that store the shared parameters for K controller replicas
  * each controller replica samples m different child architectures that are trained in parallel
  * the controller then collects gradients according to the results of that minibatch of m architectures at convergence
  * sends them to the parameter server in order to update the weights across all controller replicas
* Performance of Neural Architecture Search and other state-of-the-art models on CIFAR-10
  <image src="https://user-images.githubusercontent.com/67627471/206910176-0a7387c8-9dad-44a0-8e51-11052a09fce3.png" width=60% height=60%>
